File: Data Structures & Algorithms/products-of-array-discluding-self/submission-8.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [1 for _ in range(n)]

        for i in range(n):
            for j in range(n):
                if i != j:
                    res[i] *= nums[j]


        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        List<Integer>[] bucket = new List[nums.length+1];
        HashMap<Integer, Integer> counts = new HashMap<>();

        for(int num: nums){
            counts.put(num, counts.getOrDefault(num, 0)+1);
        }

        for (Map.Entry<Integer, Integer> entry : counts.entrySet()) {
            int idx = entry.getValue();
            if(bucket[idx]==null)bucket[idx] = new ArrayList<>();
            bucket[idx].add(entry.getKey());
        }

        int idx = 0;
        int res[] = new int[k];
        for(int i = bucket.length-1; i >= 0; i--){
            if(bucket[i] != null){
                for(int num : bucket[i]){
                    res[idx++] = num;
                    if(idx==k){
                        return res;
                    }
                }
            }
        }
        return res;

    }
}
